scripts/msct_register.py

```diff
@@ -151,34 +216,55 @@ def register2d_centermassrot(fname_src, fname_dest, fname_warp='warp_forward.nii
     # displacement_inverse = np.zeros([nz, 2])
     angle_src_dest = np.zeros(nz)
     z_nonzero = []
-    # Loop across slices
-    for iz in range(0, nz):
-        try:
-            # compute PCA and get center or mass
-            coord_src[iz], pca_src[iz], centermass_src[iz, :] = compute_pca(data_src[:, :, iz])
-            coord_dest[iz], pca_dest[iz], centermass_dest[iz, :] = compute_pca(data_dest[:, :, iz])
-            # compute (src,dest) angle for first eigenvector
-            if rot == 1:
-                eigenv_src = pca_src[iz].components_.T[0][0], pca_src[iz].components_.T[1][0]  # pca_src.components_.T[0]
-                eigenv_dest = pca_dest[iz].components_.T[0][0], pca_dest[iz].components_.T[1][0]  # pca_dest.components_.T[0]
-                angle_src_dest[iz] = angle_between(eigenv_src, eigenv_dest)
-                # check if ratio between the two eigenvectors is high enough to prevent poor robustness
-                if pca_src[iz].explained_variance_ratio_[0] / pca_src[iz].explained_variance_ratio_[1] < pca_eigenratio_th:
-                    angle_src_dest[iz] = 0
-                if pca_dest[iz].explained_variance_ratio_[0] / pca_dest[iz].explained_variance_ratio_[1] < pca_eigenratio_th:
-                    angle_src_dest[iz] = 0
-            # append to list of z_nonzero
-            z_nonzero.append(iz)
-        # if one of the slice is empty, ignore it
-        except ValueError:
-            sct.printv('WARNING: Slice #' + str(iz) + ' is empty. It will be ignored.', verbose, 'warning')
+
+    if im_and_seg is False:
```

Comment:
One problem here is that the "if case" acts on the input type rather than on the algorithm (supposed to be the latter).
One possible solution would be to use the input param `rot`, and add the entry "2" corresponding to the new method.

In the API (e.g., register_multimodal), the user could select a new flag: `-param rot_method={'pca', 'hog'}` with default=`hog`